lib/pure/os.nim

```diff
@@ -338,6 +341,62 @@ proc expandFilename*(filename: string): string {.rtl, extern: "nos$1",
       result = $r
       c_free(cast[pointer](r))
 
+proc normalizePath*(path: var string, separator=DirSep) {.rtl, extern: "nos$1", tags: [].} =
+  ## Normalize a path.
+  ##
+  ## Consecutive directory separators are collapsed, including an initial double slash.
+  ##
+  ## On relative paths, double dot (..) sequences are collapsed if possible.
+  ## On absolute paths they are always collapsed.
+  ##
+  ## Warning: URL-encoded and Unicode attempts at directory traversal are not detected.
+  ## Triple dot is not handled.
+  let is_abs = path[0] == separator
+  var stack: seq[string] = @[]
+  for p in split(path, {separator}):
+    case p
+    of "", ".":
+      continue
+    of "..":
+      if stack.len == 0:
+        if is_abs:
+          discard  # collapse all double dots on absoluta paths
+        else:
+          stack.add(p)
+      elif stack[^1] == "..":
+        stack.add(p)
+      else:
+        discard stack.pop()
+    else:
+      stack.add(p)
+
+  if is_abs:
+    path = separator & join(stack, $separator)
+  elif stack.len > 0:
+    path = join(stack, $separator)
+  else:
+    path = "."
+
+proc normalizedPath*(path: string, separator=DirSep): string {.rtl, extern: "nos$1", tags: [].} =
```

Comment:
Can you remember where in the stdlib currently this convention is in use? It's about time this is documented but I can't find any other module that uses it.